### PR TITLE
Replace DllImport with LibraryImport for WNetGetConnection

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Contributes to #18553.

- Move constants to new file since they are used for some pinvokes and we will add more in follow PRs.
- Remove intermediate method
- Remove Windows code from compiling on Unix
- Remove duplicate code
- Replace DllImport with LibraryImport

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/Interop/Windows/Errors.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+internal static partial class Interop
+{
+    internal static partial class Windows
+    {
+        // List of error constants https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes
+        internal const int ERROR_SUCCESS = 0;
+        internal const int ERROR_GEN_FAILURE = 31;
+        internal const int ERROR_NO_NETWORK = 1222;
+
+        // ERROR_INSUFFICIENT_BUFFER
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Interop/Windows/Errors.cs:15]: The line is comment for next line.

File: src/System.Management.Automation/engine/Interop/Windows/WNetGetConnection.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            uncPath = null;
+            if (s_WNetApiNotAvailable)
+            {
+                return -1;
+            }
+
+            uint bufferSize = MAX_PATH;
+
+    #if DEBUG
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Interop/Windows/WNetGetConnection.cs]: Done.

File: src/System.Management.Automation/engine/Interop/Windows/WNetGetConnection.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        [LibraryImport("mpr.dll", EntryPoint = "WNetGetConnectionW")]
+        internal static partial int WNetGetConnection(ReadOnlySpan<ushort> localName, Span<ushort> remoteName, ref uint remoteNameLength);
+
+        internal static int GetUNCForNetworkDrive(char drive, out string? uncPath)
+        {
+            uncPath = null;
+            if (s_WNetApiNotAvailable)
+            {
+                return -1;
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Interop/Windows/WNetGetConnection.cs]: Done.

File: src/System.Management.Automation/engine/Interop/Windows/Errors.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/Interop/Windows/Errors.cs:16]: So you mean `ERROR_MORE_DATA` == `ERROR_INSUFFICIENT_BUFFER`?  If so, then it makes sense, but perhaps the comment needs more context like:

```csharp
// ERROR_MORE_DATA here is equivalent to ERROR_INSUFFICIENT_BUFFER for the DEBUG path
```

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Interop/Windows/Errors.cs:16]: Yes.
`ERROR_MORE_DATA` is in docs, `ERROR_INSUFFICIENT_BUFFER` is used by .Net team (perhaps this is in SDK itself?).

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Interop/Windows/Errors.cs]: I removed the comment at all since it doesn't add understanding.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**